### PR TITLE
Add support for Rails 5.1

### PR DESCRIPTION
**Pull request by aried3r**
I guess this will need some more changes, but I'd love to see how this gem behaves with 5.1 because I'm planning to use it and upgrade our app to 5.1 soon. :)

### CODE REVIEW

**Comment by aried3r** [comment]
I think the tests with 5.1 fails because in [this commit to Rails](https://github.com/rails/rails/commit/496d744fa31665de810b404de968ba86ed87c319) a parameter has been added to `ActionController::TestCase.create`.

Also `ActionController::TestCase` has been deprecated and will be moved to its own gem in 5.1: https://github.com/rails/rails/commit/f42ac6a06880bfc02857ab71758214ad7867c22f

How would you like to proceed for this case?

**Comment by aried3r** [comment]
So, sorry for all the rebases. I hope you didn't get a message for all of them in campfire.

Anyway, some of the test runs fail because they depend on changed made in #53 wrt nokogiri and different Ruby versions.

**Comment by aried3r** [comment]
As for `/home/travis/build/plataformatec/mail_form/test/resource_test.rb:73` which fails with Rails 5.1.0.rc2 I did not yet find out why it fails.

**Comment by aried3r** [comment]
Yep, it's https://github.com/rails/rails/commit/3a25cdca3e0d29ee2040931d0cb6c275d612dffe.

Before that commit it's only a deprecation warning.
```
DEPRECATION WARNING: Returning `false` in Active Record and Active Model callbacks will not implicitly halt a callback chain in Rails 5.1. To explicitly halt the callback chain, please use `throw :abort` instead. (called from deliver at /Users/arieder/Development/mail_form/lib/mail_form/shim.rb:49)
```

I fixed it, but I'm not sure if it's the most elegant way.

**Comment by aried3r** [comment]
Rebased on master after #53 was merged. Green test run ✅

File: lib/mail_form/delivery.rb [hunk 1]
```diff
@@ -15,7 +15,12 @@ module Delivery
       class_attribute :mail_appendable
       self.mail_appendable = []
 
-      if respond_to?(:before_deliver) && respond_to?(:after_deliver)
+      # not_spam? simply returns false, but when used as a callback, we have to throw(:abort)
+      # to halt the callback chain, beginning with Rails 5.1.
+      if ActionPack.respond_to?(:version) && ActionPack.version >= Gem::Version.new('5.1')
```
Comment (rafaelfranca) [inline lib/mail_form/delivery.rb]: When would `ActionPack` not respond to `version`?

Comment (aried3r) [inline lib/mail_form/delivery.rb]: mail_form supports Rails 3.2. It didn't have `version` back then: https://github.com/rails/rails/blob/3-2-stable/actionpack/lib/action_pack/version.rb

**Pull request merged**